### PR TITLE
feat: add button and anchor as web component

### PR DESCRIPTION
**Pull request by chrisdholt**
# Description
This PR adds `fast-button` and `fast-anchor` as web components.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components/src/button/anchor.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                @slotchange=${x => x.handleEndContentChange()}
+            ></slot>
+        </span>
+    </a>
+`;
+
+export class Anchor extends FastElement {
+    @attr
+    public appearance: AnchorAppearance;
+    public appearanceChanged(): void {
```
Comment (nicholasrice) [inline packages/fast-components/src/button/anchor.ts:49]: this method gets prev and next values: `public appearanceChanged(prev: string, next: string)` so I don't think you need to track a private currentAppearance

File: packages/fast-components/src/button/button.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                name="end"
+                ${ref("end")}
+                @slotchange=${x => x.handleEndContentChange()}
+            ></slot>
+        </span>
+    </button>
+`;
+
+export class Button extends FastElement {
+    @attr
```
Comment (nicholasrice) [inline packages/fast-components/src/button/button.ts:52]: Buttons are technically form-associated elements - spec has details on implementation with that in mind.

File: packages/fast-components/src/button/anchor.ts (hunk 1 above)
Comment (chrisdholt) [reply packages/fast-components/src/button/anchor.ts:49]: Sweet, I’ll update to use the API.

File: packages/fast-components/src/button/button.ts (hunk 2 above)
Comment (chrisdholt) [reply packages/fast-components/src/button/button.ts:52]: Sounds good, I likely looked right past that. I’ll update.

File: packages/fast-components/src/styles/patterns/button.ts [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    ${display("inline-block")} :host {
+        font-family: var(--body-font);
+        outline: none;
+
+        --fast-button-corner-radius: calc(var(--corner-radius) * 1px);
+        --fast-button-fill-rest: var(--neutral-fill-rest);
+        --fast-button-fill-hover: var(--neutral-fill-hover);
+        --fast-button-fill-active: var(--neutral-fill-active);
+
+        --fast-button-foreground-rest: var(--neutral-foreground-rest);
```
Comment (nicholasrice) [inline packages/fast-components/src/styles/patterns/button.ts:15]: Are these custom properties carry-over from the initial private implementation? I thought we landed that we didn't want to get heavily involved with component-specific custom properties as an extensibility point, instead preferring selector based overrides -`fast-button.accent {}`, `fast-button::part(end) {}`

File: packages/fast-components/src/styles/patterns/button.ts [hunk 4]
```diff
@@ -0,0 +1,235 @@
+import { css } from "@microsoft/fast-element";
+import { display } from "../display";
+import { focusVisible } from "../focus";
+
+export const BaseButtonStyles = css`
+    ${display("inline-block")} :host {
+        font-family: var(--body-font);
+        outline: none;
```
Comment (nicholasrice) [inline packages/fast-components/src/styles/patterns/button.ts:8]: You probably need to rebase - the design-system implementation went in so you'll need to declare which recipes this component depends on.

**Comment by chrisdholt** [comment]
Closing to reimplement based on latest design system updates.

**Pull request closed without merging**